src/index.ts

```diff
@@ -0,0 +1,124 @@
+import { isAnyArray } from 'is-any-array';
+import BaseRegression, {
+  checkArrayLength,
+  maybeToPrecision,
+} from 'ml-regression-base';
+
+import { predict } from './utilities/predict';
+import { robustPolynomial } from './utilities/robustPolynomial';
+
+/**
+ * @class RobustPolynomialRegression
+ * @param x
+ * @param y
+ * @param degree - polynomial degree
+ */
+export default class RobustPolynomialRegression extends BaseRegression {
+  public name?: string;
+  public degree?: number;
+  public powers?: number[];
+  public coefficients?: number[];
+  public constructor(x: number[] | LoadJSON, y?: number[], degree?: number) {
```

Comment:
I would prefer that we make 3 breaking changes:

- the package should not export a 'default' so that we need to import the class using  `import {RobustPolynomialRegression} from ...` in the future
- That when we create an instance we use a https://github.com/cheminfo/cheminfo-types/blob/main/src/core/DataXY.d.ts object (and not 2 variables). Also that the option is in an option object. So it will become `const regression = new Regression( {x:[], y:[|}, {degree: 3})
- That if we want to reload a function we use `RobutstPolynomialRegression.fromModel()`

@targos WDYT ?

Reply:
> That when we create an instance we use a https://github.com/cheminfo/cheminfo-types/blob/main/src/core/DataXY.d.ts object (and not 2 variables). Also that the option is in an option object. So it will become `const regression = new Regression( {x:[], y:[|}, {degree: 3})

If you want to do this, please update ALL regression packages.

Reply:
> That if we want to reload a function we use RobutstPolynomialRegression.fromModel()

If you want to do this, please update ALL mljs packages that export a model.

Reply:
Ok indeed everything should be changed at once so we will keep this for a future refactor.